lms/djangoapps/commerce/api/v1/serializers.py

```diff
@@ -26,11 +29,31 @@ class CourseSerializer(serializers.Serializer):
     """ Course serializer. """
     id = serializers.CharField()  # pylint: disable=invalid-name
     name = serializers.CharField(read_only=True)
+    verification_deadline = serializers.DateTimeField(blank=True)
     modes = CourseModeSerializer(many=True, allow_add_remove=True)
 
+    def validate(self, attrs):
+        """ Ensure the verification deadline occurs AFTER the course mode enrollment deadlines. """
+        verification_deadline = attrs.get('verification_deadline', None)
+
+        if verification_deadline:
+            upgrade_deadline = datetime.max.replace(tzinfo=pytz.utc)
+
+            # Find the earliest upgrade deadline
+            for mode in attrs['modes']:
+                expires = mode.expiration_datetime
+                if expires:
+                    upgrade_deadline = min(expires, upgrade_deadline)
+
+            if upgrade_deadline is not None and verification_deadline < upgrade_deadline:
```

Comment:
Will `upgrade_deadline` ever be `None` at this point, if we're initializing it to `datetime.max`?

This is kind of important for the professional-with-verification case, where the mode never expires (to avoid letting users into the course for free) but it has a verification deadline.